pkg/controller/machine-set-boot-image/machine_set_boot_image_controller.go

```diff
@@ -455,6 +465,7 @@ func (ctrl *Controller) patchMachineSet(oldMachineSet, newMachineSet *machinev1b
 	if err != nil {
 		return fmt.Errorf("unable to patch new machineset: %w", err)
 	}
+	klog.Infof("Successfully patched machineset %s", oldMachineSet.Name)
```

Comment:
It's not on top of my head right now, should we explicitly reset value here to clear the failure on successful update ?

Reply:
This is a decent spot for it too! I currently have the inc/reset operations in the `handleErr()` callback - so in the future if we were to add different sync functions for other machine resources we can try to handle the metrics in a common place, if that makes sense.